### ISSUE TITLE
Clarifications on license

### ISSUE THREAD

**Issue by falkTX** [lindenbergresearch/LRTRack#65]
Hi there!

I would love to use your modules on other opensource stuff, but have some questions regarding your license terms.
On the `plugin.json` file you mention the license of the project to be `CC0-1.0` which is basically "do what you want" public domain.

But your `LICENSE` has the following text:

> Commercial redistribution of the code, or parts, in any form must be granted by the author

This is an issue, open-source code licenses never restrict commercial use.
As-is, your code might not even be suitable for hosting on GitHub since GitHub is a for-profit company.
You could say you gave rights to GitHub for hosting when you push, but this invalidates re-hosting your code in other for-profit git instances (such as bitbucket, gitlab, etc)

I do not think your code can be reused in any opensource project, but would be happy to be shown otherwise.
If really not possible, I understand, but wanted to ask just in case.

**Comment by lindenbergresearch** [comment]
Hi!

First of all the plugin.json was not proper configured, it was just the default value.
The correct license could be obtained from the LICENSE file.

I decided to switch to closed source 2 years ago because a lot of stuff took many hours of research and I want to prevent that my code is used in commercial products. 
A license of the current version  (which is hosted private on gitlab) found in VCV Rack can be obtained commercially - I already sold it to miRack.

I also decided to keep the github repo as open source, because it was open source at that time.
So in general you could use the code as found here taking care of the LICENSE.

What open source stuff do you mean?

Cheers,
Patrick

**Comment by falkTX** [comment]
> I also decided to keep the github repo as open source, because it was open source at that time. So in general you could use the code as found here taking care of the LICENSE.

Your current license on this repository is **not** open-source though.
Restricting commercial use makes it "source available", not "open source" in the terms defined by the [open source initiative](https://opensource.org/osd)

> Open source doesn't just mean access to the source code. The distribution terms of open-source software must comply with the following criteria:
>
> Free Redistribution
>
> The license shall not restrict any party from selling or giving away the software as a component of an aggregate software distribution containing programs from several different sources. The license shall not require a royalty or other fee for such sale.

> What open source stuff do you mean?

In particular https://github.com/DISTRHO/Cardinal, but later on exposing individual Rack modules and their own standalone plugins (most likely only in LV2 format, since VST does not have proper CV port support)

**Comment by lindenbergresearch** [comment]
Ahh, ok, I see what you mean. I am not an expert on this. ;-) 
You can use the code, no problem, I like your project!
As long as you keep the name and logo, but that's given as far as I understood.

**Comment by falkTX** [comment]
While I appreciate your thoughts, your license is incompatible with GPL so I cannot use any of your code.
If you relicense to any usual opensource license (BSD, GPL, apache, MPL, etc etc) than this could be allowed..

As I kinda mentioned above, per your code license at the moment, even rehosting would be illegal.

**Comment by lindenbergresearch** [comment]
It should be OK now.

**Comment by falkTX** [comment]
So you are ok with commercial use after all?

I would kindly ask you to change `plugin.json` then, to reflect the changes.
The correct string would be `GPL-3.0-or-later` as defined by spdx.dev

Thanks a lot!

**Comment by lindenbergresearch** [comment]
Maybe I didn't understand you correct - you can use it for open-source project, not for commercial stuff.
For commercial usage I sell license as I did for miRack for example. It would be not fair to sell stuff you didn't made :-) 
Wasn't your project open-source?

**Comment by falkTX** [comment]
I think you are not understanding.

opensource and commercial are not antonyms.
you are thinking of closed-source.

there is nothing forbidding an opensource project to be commercial in nature. restricting commercial usage makes a project NOT opensource.
it is within the GPL to sell or do whatever you want with the software **but you have to share the source code modifications under the same license**

The same idea applies to Creative Commons Share-Alike license. One can freely use for anything, even selling, as long as they share any modifications within the same license.

Basically, if you restrict commercial use, your project cannot be called opensource.
You can restrict closed-source usage though, and that is what GPL was created for. Other licenses like BSD/MIT/ISC allow closed-source usage though.

**Comment by dromer** [comment]
> Maybe I didn't understand you correct - you can use it for open-source project, not for commercial stuff. For commercial usage I sell license as I did for miRack for example. It would be not fair to sell stuff you didn't made :-) Wasn't your project open-source?

This would be a dual license, which you should very clearly need to mark the project as. Dual licensed software is quite problematic though, it comes down to what @falkTX described as "source available", but not actually compatible with opensource licensing.

**Comment by falkTX** [comment]
Licenses are a serious thing, the Cardinal project **must** have all code GPLv3+ compatible.
I opened this ticket just to see if it was possible. No hard feelings if not.

PS: The use of open-source and commercial can be seen in several places. https://ardour.org/ is the most well known for audio folks on Linux, where the source code is free but official binaries are behind a pay-wall. It helps to support the project and keep the lights on. Has been working well for them for many, many years now. Basically paying for the convenience of ready-to-go binaries, but you are free to build yourself of course.

If I ever submit any modules to VCV Rack library I will follow the same approach: source code is 100% free to do what you want. Relying on donations doesn't really work, as most people never donate anything. The convenience of official online library integration is in my opinion a reasonable thing to pay for.

**Comment by lindenbergresearch** [comment]
OK, thanks for the clarification, I think I understood now.
Let me think about it, I am kinda busy right now... thanks

**Comment by lindenbergresearch** [comment]
Sounds strange to me, sorry. I don't change anything and set back the original license.
If you want to use my code in any form, you pleased to buy a license. Thanks.

**Comment by falkTX** [comment]
If I buy a license, can I make it fully & proper opensource?

**Comment by lindenbergresearch** [comment]
If you buy a license I change the license to GPL - for the repository as it is - and you could do what you want to. Is that your intention?
 What do you offer?

**Comment by falkTX** [comment]
Hah I dont know yet, depends on how much a crowfund campaign would get.
Was mostly asking if that was even a possibility.

I am quite poor myself, doing free&open source software is not something that makes me rich :sweat_smile:

**Comment by lindenbergresearch** [comment]
Well maybe you should get a better job ;-) 
Just to get an idea, I sold last year a license for an iOS Project for around $2k.
So if you offer at lease $1k I will be fine...

**Comment by falkTX** [comment]
We all have our  own targets, I dont mind skipping on luxuries if that allows me to do better things.

In any case, I will keep your proposal in mind, thank you.
